### PR TITLE
Revision to Sizing:Zone Error Message

### LINKED ISSUE #8827: Limit of Zone Cooling Design Supply Air Temperature in Sizing:Zone object
Issue overview
--------------
It's nowhere that the Zone Cooling Design Supply Air Temperature input is given a range. But if this input is negative, say -20°C, the simulation returns an error message saying this input should not be negative.

    _Severe ** Sizing:Zone="OfficeZone", invalid data.
    ~~~ ** ... incorrect Zone Cooling Design Supply Air Temperature=[-20.00], value should not be negative.
    Fatal ** Sizing:Zone: Errors found in getting input. Program terminates._

Is it intended that the minimum input should be 0? If yes, please update the documentation (including IDD) to reflect this. Otherwise, maybe the input range needs to extend to take negative value.

### Some additional details for this issue:
 - Platform: Windows 7 Professional
 - Version of EnergyPlus: 9.4.0

### PR DESCRIPTION
**Pull request by RKStrand**
Pull request overview
---------------------
 - Fixes #8827 
 - This pull request provides code, unit tests, and modified documentation to fix a problem with the error messages for the supply air temperatures for the zone sizing object.  When the object was read in, there was a built in assumption that neither the heating or cooling supply air temperature was negative.  This was simply a test to avoid bad input.  While "reasonable", this was never documented and led to a severe error.  While very unlikely, it is possible that the values could be outside of the normal range.  It was suggested that the logic be modified so that the error message was only a warning with a note to double check the input values.  This was done and in addition it was verified that the heating supply air temperature is also greater than the cooling supply air temperature and that a severe error results if this is not the case.  A new unit test was created to test the new subroutine that produces the new error messages.  In addition, the documentation was enhanced so that the user knows that a warning will be produced if the supply air temperatures are negative and a severe error will be produced if the heating temperature is lower than the cooling temperature.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: doc/input-output-reference/src/overview/group-design-objects.tex [hunk 1]
```diff
@@ -1087,7 +1087,7 @@ \subsubsection{Inputs}\label{inputs-4-008}
 
 \paragraph{Field: Zone Cooling Design Supply Air Temperature}\label{field-zone-cooling-design-supply-air-temperature}
 
-The supply air temperature in degrees Celsius for the zone cooling design air flow rate calculation. Air is supplied to the zone at this temperature during the cooling design day simulation, The zone load is met by varying the zone air flow rate. The maximum zone flow rate is saved as the~ zone cooling design air flow rate. This field is only used when Zone Cooling Design Supply Air Temperature Input Method = \textbf{SupplyAirTemperature}.
+The supply air temperature in degrees Celsius for the zone cooling design air flow rate calculation. Air is supplied to the zone at this temperature during the cooling design day simulation. The zone load is met by varying the zone air flow rate. The maximum zone flow rate is saved as the~ zone cooling design air flow rate. This field is only used when Zone Cooling Design Supply Air Temperature Input Method = \textbf{SupplyAirTemperature}. If the value entered for this parameter is less than zero, a warning message is produced so that the user will double check the input to make sure it is correct.
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-design-objects.tex:1090]: Reasonable.

File: src/EnergyPlus/SizingManager.cc [hunk 2]
```diff
@@ -3373,6 +3360,32 @@ void GetZoneSizingInput(EnergyPlusData &state)
     }
 }
 
+void ReportTemperatureInputError(
```
Comment (Myoldmopar) [inline src/EnergyPlus/SizingManager.cc:3363]: At first I wasn't quite sure why you were passing a zero in here, but this makes good sense now.  Nice reuse of the new worker function.

File: tst/EnergyPlus/unit/SizingManager.unit.cc [hunk 3]
```diff
@@ -540,3 +541,54 @@ TEST_F(EnergyPlusFixture, SizingManager_CalcdoLoadComponentPulseNowTest)
...
+
+    // Test 3: Heating Temperature is lower Cooling Temperature--this should become a severe error
+    state->dataIPShortCut->rNumericArgs(paramNum) = -2.0;
+    ReportTemperatureInputError(*state, objectName, paramNum, lowTempLimit, true, errorsFound);
+    ASSERT_TRUE(errorsFound);
+    EXPECT_TRUE(compare_err_stream(delimited_string({"   ** Severe  ** Sizing:Zone=\"SPACE1-1\" has invalid data.",
+                                                     "   **   ~~~   ** ... incorrect Zone Heating Design Supply Air Temperature=[-2.00] is less than "
+                                                     "Zone Cooling Design Supply Air Temperature=[-1.00]",
+                                                     "   **   ~~~   ** This is not allowed.  Please check and revise your input."})));
+}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/SizingManager.unit.cc:594]: Nice.

**Review by Myoldmopar — APPROVED** [review]
All good here.

**Comment by Myoldmopar** [comment]
I didn't see a defect file on the PR, the issue, or in DevSupport.  The unit tests provide very good coverage, however, and the functionality being pulled into a small worker function makes it easy to understand the changes.  This is good stuff.  I pulled in develop and ran the full test suite, all good.  CI is all happy as well.  This is merging.  Thanks @RKStrand

File: src/EnergyPlus/SizingManager.cc [hunk 4]
```diff
@@ -3373,6 +3360,32 @@ void GetZoneSizingInput(EnergyPlusData &state)
...
+    EnergyPlusData &state, std::string cObjectName, int const paramNum, Real64 comparisonTemperature, bool const shouldFlagSevere, bool &ErrorsFound)
+{
+    if (state.dataIPShortCut->rNumericArgs(1) < comparisonTemperature) {
+        if (shouldFlagSevere) { // heating supply air temperature is lower than cooling supply air temperature--not allowed
```
Comment (mjwitte) [inline src/EnergyPlus/SizingManager.cc:3367]: @RKStrand I just encountered this error with a user's file. This seems overly strict. A warning would be fine, but halting the simulation seems overly strict. The only important thing is that the heating supply temp is greater than the heating setpoint. And likewise for cooling. 
```
   ** Severe  ** Sizing:Zone="C00138" has invalid data.
   **   ~~~   ** ... incorrect Zone Heating Design Supply Air Temperature=[21.00] is less than Zone Cooling Design Supply Air Temperature=[24.00]
   **   ~~~   ** This is not allowed.  Please check and revise your input.
```

Comment (RKStrand) [reply src/EnergyPlus/SizingManager.cc:3367]: Hmmm...I don't remember the specifics anymore as to why this was made a severe.  In looking back at the issue, it seems like you advocated making this a warning.  There are some other cases that are warnings.  I'm not sure why this was made a "severe" except that it seems illogical to have a heating supply air temperature that is less than a cooling supply air temperature.  I must not have foreseen this as a possibility.  @Myoldmopar how should we proceed here?

Comment (mjwitte) [reply src/EnergyPlus/SizingManager.cc:3367]: @RKStrand Given that the original issues was about supply temps < 0, this check of heating > cooling supply temp added an additional constraint. I agree it typically might not make sense, so I would be fine with a warning. I will post a new issue for this.

**Pull request merged**